<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>JointSpaceVisualizer</class>
 <widget class="qMRMLWidget" name="JointSpaceVisualizer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="maxillaLabel">
        <property name="text">
         <string>Target Model (e.g., Maxilla):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QWidget" name="maxillaSelectorContainer" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_1">
         <item>
          <widget class="qMRMLNodeComboBox" name="maxillaSelector">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLModelNode</string>
            </stringlist>
           </property>
           <property name="noneEnabled">
            <bool>true</bool>
           </property>
           <property name="showChildNodeTypes">
            <bool>false</bool>
           </property>
           <property name="addEnabled">
            <bool>false</bool>
           </property>
           <property name="removeEnabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="maxillaLoadButton">
           <property name="text">
            <string>Load…</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="mandibleLabel">
        <property name="text">
         <string>Source Model (e.g., Mandible):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QWidget" name="mandibleSelectorContainer" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="qMRMLNodeComboBox" name="mandibleSelector">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLModelNode</string>
            </stringlist>
           </property>
           <property name="noneEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="mandibleLoadButton">
           <property name="text">
            <string>Load…</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <widget class="ctkCollapsibleButton" name="decimationCollapsibleButton">
     <property name="text">
      <string>Decimation Options</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="decimationFormLayout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="enableDecimationCheckBox">
        <property name="text">
         <string>Enable Mesh Decimation</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="decimationLabel">
        <property name="text">
         <string>Target Reduction (%):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLSliderWidget" name="decimationSlider">
        <property name="minimum">
         <double>0.0</double>
        </property>
        <property name="maximum">
         <double>99.0</double>
        </property>
        <property name="value">
         <double>90.0</double>
        </property>
        <property name="singleStep">
         <double>1.0</double>
        </property>
        <property name="toolTip">
         <string>Target percentage of points to remove from the mesh.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="applyButton">
     <property name="toolTip">
      <string>Run the distance calculation.</string>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>

   <item>
    <widget class="ctkCollapsibleButton" name="displayCollapsibleButton">
     <property name="text">
      <string>Display</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="displayFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="scaleLabel">
        <property name="text">
         <string>Scale (fixed):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="scaleInfoLabel">
        <property name="text">
         <string>0–1.0 mm: Red → 2.5 mm: Yellow → 3.25 mm: Green → 4–5 mm: Blue</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="minDistanceLabel">
        <property name="text">
         <string>Min Distance (mm):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="minDistanceValueLabel">
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="resultLabel">
        <property name="text">
         <string>Result (DistanceMap)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
        <widget class="QWidget" name="resultControls" native="true">
         <layout class="QHBoxLayout" name="resultControlsLayout">
          <item>
           <widget class="QCheckBox" name="resultVisibilityCheckBox">
           <property name="text">
            <string>Show</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="qMRMLSliderWidget" name="resultOpacitySlider">
           <property name="minimum">
            <double>0.0</double>
           </property>
           <property name="maximum">
            <double>100.0</double>
           </property>
           <property name="value">
            <double>100.0</double>
           </property>
           <property name="singleStep">
            <double>1.0</double>
           </property>
           <property name="toolTip">
            <string>Result opacity (%)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="saveResultButton">
           <property name="text">
            <string>Save Result…</string>
           </property>
           <property name="toolTip">
            <string>Export the distance-mapped model to a file</string>
           </property>
          </widget>
         </item>
         </layout>
        </widget>
      </item>

      <item row="3" column="0">
       <widget class="QLabel" name="maxillaDispLabel">
        <property name="text">
         <string>Target (Maxilla)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QWidget" name="maxillaControls" native="true">
        <layout class="QHBoxLayout" name="maxillaControlsLayout">
         <item>
          <widget class="QCheckBox" name="maxillaVisibilityCheckBox">
           <property name="text">
            <string>Show</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="qMRMLSliderWidget" name="maxillaOpacitySlider">
           <property name="minimum">
            <double>0.0</double>
           </property>
           <property name="maximum">
            <double>100.0</double>
           </property>
           <property name="value">
            <double>100.0</double>
           </property>
           <property name="singleStep">
            <double>1.0</double>
           </property>
           <property name="toolTip">
            <string>Maxilla opacity (%)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>

      <item row="4" column="0">
       <widget class="QLabel" name="mandibleDispLabel">
        <property name="text">
         <string>Source (Mandible)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QWidget" name="mandibleControls" native="true">
        <layout class="QHBoxLayout" name="mandibleControlsLayout">
         <item>
          <widget class="QCheckBox" name="mandibleVisibilityCheckBox">
           <property name="text">
            <string>Show</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="qMRMLSliderWidget" name="mandibleOpacitySlider">
           <property name="minimum">
            <double>0.0</double>
           </property>
           <property name="maximum">
            <double>100.0</double>
           </property>
           <property name="value">
            <double>100.0</double>
           </property>
           <property name="singleStep">
            <double>1.0</double>
           </property>
           <property name="toolTip">
            <string>Mandible opacity (%)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSliderWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 </ui>
